lib/encoder.py

```diff
@@ -836,29 +857,33 @@ def StoreEncoding(self, encoding, workdir=None):
   def ReadEncodingResult(self, encoding, scoredir=None):
     """Reads an encoding result back from storage, if present.
 
-    None is returned file does not exist."""
+    None is returned if file does not exist.
+    If scoredir is given, only that directory is searched.
+    If it is not given, all directories in the search path are searched."""
 
     if scoredir:
-      workdir = os.path.join(scoredir, self.context.codec.name)
+      searchpath = [os.path.join(scoredir, self.context.codec.name)]
     else:
-      workdir = self.workdir
-    dirname = ('%s/%s/%s' % (workdir, encoding.encoder.Hashname(),
-                             self.context.codec.SpeedGroup(encoding.bitrate)))
-    filename = '%s/%s.result' % (dirname, encoding.videofile.basename)
-    if os.path.isfile(filename):
-      with open(filename, 'r') as resultfile:
-        stringbuffer = resultfile.read()
-        try:
-          return json.loads(stringbuffer)
-        except exceptions.ValueError:
+      searchpath = self.SearchPath()
+
+    for workdir in searchpath:
+      dirname = ('%s/%s/%s' % (workdir, encoding.encoder.Hashname(),
+                              self.context.codec.SpeedGroup(encoding.bitrate)))
+      filename = '%s/%s.result' % (dirname, encoding.videofile.basename)
+      if os.path.isfile(filename):
```

Comment:
Mumble about whether "continue" is an improvement. The try block looked a bit better when I removed the legacy AST-reading code (I have a tool for the conversion that doesn't depend on this now.)